File: ee-schkola/src-gen/main/kotlin/ee/schkola/finance/FinanceApiBase.kt
package ee.schkola.finance

import ee.schkola.SchkolaBase
import ee.schkola.person.Profile
import java.util.*


open class Expense : SchkolaBase {
    val purpose: ExpensePurpose
    val amount: Float
    val profile: Profile
    val date: Date


    constructor(id: String = "", purpose: ExpensePurpose = ExpensePurpose(), amount: Float = 0f,
        profile: Profile = Profile(), date: Date = Date()) : super(id) {
        this.purpose = purpose
        this.amount = amount
        this.profile = profile
        this.date = date
    }

    companion object {
        val EMPTY = Expense()
    }
}


open class ExpensePurpose : SchkolaBase {
    val name: String
    val description: String


    constructor(id: String = "", name: String = "", description: String = "") : super(id) {
        this.name = name
        this.description = description
    }

    companion object {
        val EMPTY = ExpensePurpose()
    }
}


open class Fee : SchkolaBase {
    val student: Profile
    val amount: Float
    val kind: FeeKind
    val date: Date


    constructor(id: String = "", student: Profile = Profile(), amount: Float = 0f, kind: FeeKind = FeeKind(),
        date: Date = Date()) : super(id) {
        this.student = student
        this.amount = amount
        this.kind = kind
        this.date = date
    }

    companion object {
        val EMPTY = Fee()
    }
}


open class FeeKind : SchkolaBase {
    val name: String
    val amount: Float
    val description: String


    constructor(id: String = "", name: String = "", amount: Float = 0f, description: String = "") : super(id) {
        this.name = name
        this.amount = amount
        this.description = description
    }

    companion object {
        val EMPTY = FeeKind()
    }
}

File: ee-schkola/src-gen/main/go/ee/schkola/finance/FinanceApiBase.go
package finance

import (
    "ee/schkola"
    "ee/schkola/person"
    "time"
)

type Expense struct {
    Purpose  *ExpensePurpose
    Amount  float64
    Profile  *person.Profile
    Date  *time.Time
    *schkola.SchkolaBase
}

func NewExpense(purpose *ExpensePurpose, amount float64, profile *person.Profile, date *time.Time, SchkolaBase *schkola.SchkolaBase) (ret *Expense) {
    ret = &Expense{
        Purpose : purpose,
        Amount : amount,
        Profile : profile,
        Date : date,
        SchkolaBase: SchkolaBase,
    }
    
    return
}



type ExpensePurpose struct {
    Name  string
    Description  string
    *schkola.SchkolaBase
}

func NewExpensePurpose(name string, description string, SchkolaBase *schkola.SchkolaBase) (ret *ExpensePurpose) {
    ret = &ExpensePurpose{
        Name : name,
        Description : description,
        SchkolaBase: SchkolaBase,
    }
    
    return
}



type Fee struct {
    Student  *person.Profile
    Amount  float64
    Kind  *FeeKind
    Date  *time.Time
    *schkola.SchkolaBase
}

func NewFee(student *person.Profile, amount float64, kind *FeeKind, date *time.Time, SchkolaBase *schkola.SchkolaBase) (ret *Fee) {
    ret = &Fee{
        Student : student,
        Amount : amount,
        Kind : kind,
        Date : date,
        SchkolaBase: SchkolaBase,
    }
    
    return
}



type FeeKind struct {
    Name  string
    Amount  float64
    Description  string
    *schkola.SchkolaBase
}

func NewFeeKind(name string, amount float64, description string, SchkolaBase *schkola.SchkolaBase) (ret *FeeKind) {
    ret = &FeeKind{
        Name : name,
        Amount : amount,
        Description : description,
        SchkolaBase: SchkolaBase,
    }
    
    return
}









